src/v1/models/ansible.js

```diff
@@ -107,4 +107,78 @@ export default class AnsibleModel extends KubeModel {
       job: _.get(au, 'status.k8sJob.namespacedName'),
     }));
   }
+
+  async createAndUpdateAnsibleJobs(args) {
+    const { toCreateJSON, toUpdateJSON } = args;
+    const createRes = [];
+    const createErr = [];
+    if (Array.isArray(toCreateJSON) && toCreateJSON.length > 0) {
+      const ur = await Promise.all(toCreateJSON.map((json) => this.ansibleJobsAction(json, 'post')
+        .then((res) => ({ response: res, kind: json.kind }))
+        .catch((err) => ({ status: 'Failure', message: err.message, kind: json.kind }))));
+      ur.forEach((item) => {
+        if (item.status === 'Failure' || item.message) {
+          createErr.push({
+            message: item.message,
+            kind: item.kind,
+          });
+        } else {
+          createRes.push({
+            response: item.response ? item.response : item,
+            kind: item.kind,
+          });
+        }
+      });
+    }
+    const updateRes = [];
+    const updateErr = [];
+    if (Array.isArray(toUpdateJSON) && toUpdateJSON.length > 0) {
+      const ur = await Promise.all(toUpdateJSON.map((json) => this.ansibleJobsAction(json, 'put')
+        .then((res) => ({ response: res, kind: json.kind }))
+        .catch((err) => ({ status: 'Failure', message: err.message, kind: json.kind }))));
+      ur.forEach((item) => {
+        if (item.status === 'Failure' || item.message) {
+          updateErr.push({
+            message: item.message,
+            kind: item.kind,
+          });
+        } else {
+          updateRes.push({
+            response: item.response ? item.response : item,
+            kind: item.kind,
+          });
+        }
+      });
+    }
+    return {
+      create: {
+        errors: createErr,
+        result: createRes,
+      },
+      update: {
+        errors: updateErr,
+        result: updateRes,
+      },
+    };
+  }
+
+  async ansibleJobsAction(action, json) {
+    const namespace = _.get(json, 'metadata.namespace');
+    const url = `/apis/tower.ansible.com/v1alpha1/namespaces/${namespace}/ansiblejobs`;
```

Comment:
this doesn't look correct to me. You shouldn't need to modify ansiblejob. The UI should only modify PolicyAutomation CR

Reply:
The API is wrong. it should be `/apis/policy.open-cluster-management.io/v1beta1/namespaces/${namespace}/policyautomations`